Wrappers/Python/cil/framework/framework.py

```diff
@@ -1795,25 +1808,70 @@ class DataContainer(object):
     '''Generic class to hold data
     
     Data is currently held in a numpy arrays'''
-    
+
+    @property
+    def geometry(self):
+        return None
+
+    @geometry.setter
+    def geometry(self, val):
+        if val is not None:
+            raise TypeError("DataContainers cannot hold a geometry, use ImageData or AcquisitionData instead")
+
+    @property
+    def dimension_labels(self):
+        default_labels = [0]*self.number_of_dimensions
+        for i in range(self.number_of_dimensions):
+            default_labels[i] = 'dimension_{0:02}'.format(i)
+
+        if self.__dimension_labels is None:
+            return tuple(default_labels)
+        else:
+            return self.__dimension_labels
+      
+    @dimension_labels.setter
+    def dimension_labels(self, val):
+        if val is None:
+            self.__dimension_labels = None
+        elif len(val)==self.number_of_dimensions:
+            self.__dimension_labels = tuple(val)
+        else:
+            raise ValueError("dimension_labels expected a list containing {0} strings got {1}".format(self.number_of_dimensions, labels))
+
+    @property
+    def shape(self):
+        '''Returns the shape of the  of the DataContainer'''
+        return self.as_array().shape
```

Comment:
use the actual numpy array to get the shape as ` self.array.shape`

and the methods below as well. 

The consideration being, that this is wrapping a numpy array and when we will be using a different backend we will be providing these properties correctly without needing to pass through the numpy array.